File: v1/src/main/java/com/api/v1/medical_slots/domain/MedicalSlotAuditTrail.java
package com.api.v1.medical_slots.domain;

import com.api.v1.medical_slots.MedicalSlot;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.UUID;

@Document(collection = "MedicalSlotAuditTrail")
public record MedicalSlotAuditTrail(
        @Id
        String id,
        MedicalSlot medicalSlot,
        LocalDateTime createdAt
) {

    public static MedicalSlotAuditTrail of(MedicalSlot medicalSlot) {
        return new MedicalSlotAuditTrail(
                UUID.randomUUID().toString(),
                medicalSlot,
                LocalDateTime.now()
        );
    }
}

File: v2/src/main/kotlin/com/api/v2/medical_slots/MedicalSlotAuditTrail.kt
package com.api.v2.medical_slots

import org.springframework.data.annotation.Id
import java.time.LocalDateTime
import java.util.UUID

data class MedicalSlotAuditTrail(
    @Id
    val id: String,
    val medicalSlot: MedicalSlot,
    val createdAt: LocalDateTime
) {

    companion object {
        fun of(medicalSlot: MedicalSlot): MedicalSlotAuditTrail {
            return MedicalSlotAuditTrail(
                UUID.randomUUID().toString(),
                medicalSlot,
                LocalDateTime.now()
            )
        }
    }
}
